runtime/near-vm-logic/src/logic.rs

```diff
@@ -1148,15 +1155,22 @@ impl<'a> VMLogic<'a> {
         pub_key_ptr: u64,
     ) -> Result<u64> {
         use ed25519_dalek::{PublicKey, Signature, Verifier, SIGNATURE_LENGTH};
+        const MESSAGE_LENGTH: usize = 32;
 
         self.gas_counter.pay_base(ed25519_verify_base)?;
-        if sig_len != SIGNATURE_LENGTH as u64 {
+        let msg = self.get_vec_from_memory_or_register(msg_ptr, msg_len)?;
+        if msg_len != MESSAGE_LENGTH as u64 || msg.len() != MESSAGE_LENGTH {
```

Comment:
I think you've meant tocheck lenth of the public key, not the lenght of the message?